### ISSUE TITLE
UploadMedia is not working with Mono and Linux (Docker Container)

### ISSUE THREAD

**Issue by Dom80** [Yortw/tweetmoasharp#43]
At first, thank you, to bring tweetsharp back to live.
We use TweetMoaSharp and it works great with windows, but we switched our infrastructure to linux docker. 
Now, we have a problem with UploadMedia.
Twitter gives us an error back. 
The stream content object is not correct.
The problem is the hammcock library, it uses to create the stream content object an StringBuilder. This StringBuilder use Environment.Newline to add new lines.
This Newlines are different between Windows and Linux. 

I know this is a small issue, we have our workaround and .NET CORE is the new way to go. But i thought it is not a bad idea to report this issue.

**Comment by Yortw** [comment]
Thanks for reporting this issue.

So you have a workaround?

I don't have any Linux machines and am totally unfamiliar with Docket & Mono. I understand the issue, but would be nervous about trying to fix it (even though it *seems* simple) given my inability to test it. 

I'm also not sure how .Net Core helps - I believe it also uses Environment.NewLine and that *should* be different on Windows vs Linux regardless of the .Net platform?

**Comment by Dom80** [comment]
My workaround is to use HttpClient instead of Hammocks RestClient in my own Function. But you need the current .net Framework to use HttpClient.
I was to lazy to compile my own Hammock lib, sorry.

My Linux run in a virtualbox. Good enough to reproduce this problem.

With .NET CORE i am not familar, but i will switch from Mono to .NET CORE in my next API project.



```
using (var dlClient = new HttpClient())
            {
                using (var postClient = new HttpClient())
                {
                    postClient.DefaultRequestHeaders.Add("Authorization", authheader);
                    using (var content = new MultipartFormDataContent(Guid.NewGuid().ToString()))
                    {
                        content.Add(new StreamContent(dlClient.GetStreamAsync(new Uri(fileUri)).Result), "media", fileUri);
                        var resultRequest = postClient.PostAsync(url, content).Result.Content;
                        response = resultRequest.ReadAsStringAsync().Result;
                    }
                }
            }

            var result = JsonConvert.DeserializeObject<TwitterUploadedMedia>(response);
```

**Comment by Yortw** [comment]
Hi,

Spent some time looking at this last night but I couldn't find anywhere .AppendLine was used from StringBuilder, nor any references to Environment.NewLine that looked relevant. Can you give the name of the method in Hammock where you think this is happening?

Thanks.

**Comment by Yortw** [comment]
I hope this is fixed in the 3.0.0.22 release. If it is not, please let me know.
